### PR TITLE
Add search functionality request#175743332

### PR DESCRIPTION
**Pull request by ConstantineMaliza**
#### What does this PR do?

- Manager and Requester can search request

#### Description of Task to be completed? 

- To search requester must be the authenticated user or manager

#### How should this be manually tested?

- Clone the repo 

- In your home directory  cd this branch **`Search-API#175743332`**

- Then run **`npm install`** to install all dependencies

- Run **`npm start`** to start the server

- In your default browser navigate to **`http://localhost:5000`** 

#### Any background context you want to provide?

- To requester use the endpoint `get` in path **`http://localhost:5000/request/search`** 
- To manager use the endpoint `get` **`http://localhost:5000/request/search/manager`**  
- To manager  to search on requester table use the endpoint `get` **`http://localhost:5000/request/search/requester`**  
- **we will params according to the query** 

#### What are the relevant Pivotal Tracker stories?

- #175743332

#### Screenshots (if appropriate)
- N/A

#### Questions:

- N/A

### CODE REVIEW

File: src/controllers/searchController.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      else queries.push( { [key]: value })
+      
+    }
+    return queries;
+}
+
+export const searchRequest = async (req, res) => {
+    const queries = returnQueries(req.query);
+    const { id } = req.userData;
+    const Request = await model.request.findAll({
```
Comment (joshwambere) [inline src/controllers/searchController.js:23]: Try checking if the request to the database has gone successfully so you can know which error to throw if it didn't go through cz when you pass incorrect query like `?idUser=kl `the request keep going endlessly it should show bad request or another error message

**Review by joshwambere — CHANGES_REQUESTED** [review]

**Review by thepatrickniyo — APPROVED** [review]

**Review by joshwambere — APPROVED** [review]

File: src/tests/search.test.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        const role = await model.userRoles.findOne({where:{name: 'MANAGER',}});
+        const user = await model.User.create({
+            firstname: "Maliza",
+            lastname: "Constantine",
+            telephone: "",
+            email: "malizacoco4@gmail.com",
+            password: "1234567$#8",
+            gender: "Female",
+            origin: "",
+            profession: "",
```
Comment (SimonImanigirimpuhwe) [inline src/tests/search.test.js]: Remove profession field since we don't have it in user model

File: src/tests/search.test.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    it("should search request", async() => {
+        const user = await model.User.create({
+            firstname: "Maliza",
+            lastname: "Constantine",
+            telephone: "",
+            email: "malizacoco4@gmail.com",
+            password: "1234567$#8",
+            gender: "Female",
+            origin: "",
+            profession: "",
```
Comment (SimonImanigirimpuhwe) [inline src/tests/search.test.js]: The same here

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]

**Review by SimonImanigirimpuhwe — APPROVED** [review]

File: src/controllers/searchController.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import Sequelize from 'sequelize'
+
+const Op = Sequelize.Op;
+
+const returnQueries = (query) =>{
+    const queries = [];
+    for (const key in query) {
+      const value = query[key];
+      
+      if (key === 'requestStatus') queries.push({ [key]: { [Op.iLike]: `%${value}%` } }) ;
```
Comment (SamuelSabasNayo) [inline src/controllers/searchController.js]: Please try to query or search a request using all possible string fields (i.e. requesterName, hotelName, location, roomType)

**Review by SamuelSabasNayo — CHANGES_REQUESTED** [review]

**Review by SamuelSabasNayo — APPROVED** [review]

**Pull request merged**